interp/impargs.ml

```diff
@@ -690,6 +690,9 @@ let projection_implicits env p impls =
   CList.skipn_at_least npars impls
 
 let declare_manual_implicits local ref ?enriching l =
+  assert (List.for_all (List.for_all (fun (_, (max, fi, fu)) -> fi && fu)) l);
+  assert (List.for_all (List.for_all (fun (ex, _) -> match ex with ExplByPos (_,_) -> true | _ -> false)) l);
+  assert (List.length l = 1);
```

Comment:
Please enforce this length invariant by typing instead.
Not sure what the other asserts mean, if reasonable to do please also enforce by typing.